code/datums/components/mood.dm

```diff
@@ -212,7 +216,10 @@
 		amount = sanity + 0.5
 	else if(sanity > maximum && amount > sanity - 0.5)
 		amount = sanity - 0.5
-	sanity = amount
+
+	// Disturbed stops you from getting any more sane
+	if(!(owner.has_trait(TRAIT_UNSTABLE) && amount > sanity))
+		sanity = amount
```

Comment:
"If you don't have the sanity trait and your adjusted amount is higher than current sanity, set sanity to adjusted amount"

That doesn't sound right.

Really you could just remove this check and put it on the check before +0.5 addition

`if(!(owner.has_trait(TRAIT_UNSTABLE)) && sanity < minimum && amount < sanity + 0.5)`

Reply:
okay thanks, i'm never good with these

Reply:
actually that might not work if you have an amount higher than sanity hmmmm

Reply:
```DM
if(owner.has_trait(TRAIT_UNSTABLE))
    sanity = min(amount,sanity)
else
    sanity = amount
```

That might be more appropriate?

Reply:
ok that's perfect and very clear in what it does